src/Objects/_PVector.js

```diff
@@ -0,0 +1,283 @@
+module.exports = function(options, undef) {
+  'use strict';
+
+  const TAU = 2 * Math.PI,
+        //TAU = options.PConstants.TAU,
+        lerp = function(start, stop, amt) {
+          return amt * (stop - start) + start;
+        },
+        sq = function(n) {
+          return n*n;
+        },
+        argsErr = function(mtd, len, min) {
+          throw 'Too few args passed to ' + mtd +
+                '() [' + len + ' < ' + min + '].';
+        };
+
+  function _PVector(x, y, z) {
+    this.x = +x || 0;
+    this.y = +y || 0;
+    this.z = +z || 0;
+  }
+
+  _PVector.fromAngle = function(angle, target) {
+    return target? target.set(Math.cos(angle), Math.sin(angle))
+                 : new _PVector(Math.cos(angle), Math.sin(angle));
+  };
+
+  _PVector.random2D = function(target, parent) {
+    //const rnd = parent? parent : target instanceof Processing? target : Math;
+    const rnd = parent? parent : target && target.random? target : Math;
+    return _PVector.fromAngle(TAU * rnd.random(), target);
+  };
+
+  _PVector.random3D = function(target, parent) {
+    //const rnd = parent? parent : target instanceof Processing? target : Math,
+    const rnd = parent? parent : target && target.random? target : Math,
+          ang = TAU * rnd.random(),
+          vz  = 2 * rnd.random() - 1,
+          vzr = Math.sqrt(1 - vz*vz),
+          vx  = vzr * Math.cos(ang),
+          vy  = vzr * Math.sin(ang);
+    return target? target.set(vx, vy, vz) : new _PVector(vx, vy, vz);
+  };
+
+  _PVector.dist = function(v1, v2) {
+    return Math.sqrt(_PVector.distSq(v1, v2));
+  };
+
+  _PVector.distSq = function(v1, v2) {
+    return sq(v1.x - v2.x) + sq(v1.y - v2.y) + sq(v1.z - v2.z);
+  };
+
+  _PVector.dot = function(v1, v2) {
+    return v1.x*v2.x + v1.y*v2.y + v1.z*v2.z;
+  };
+
+  _PVector.cross = function(v1, v2, target) {
+    const cx = v1.y*v2.z - v2.y*v1.z,
+          cy = v1.z*v2.x - v2.z*v1.x,
+          cz = v1.x*v2.y - v2.x*v1.y;
+    return target? target.set(cx, cy, cz) : new _PVector(cx, cy, cz);
+  };
+
+  _PVector.angleBetween = function(v1, v2) {
+    if (!v1.x && !v1.y && !v1.z || !v2.x && !v2.y && !v2.z) { return 0; }
+    const amt = _PVector.dot(v1, v2) / Math.sqrt(v1.magSq() * v2.magSq());
+    return amt <= -1? Math.PI : amt >= 1? 0 : Math.acos(amt);
+  };
+
+  _PVector.lerp = function(v1, v2, amt) {
+    return v1.copy().lerp(v2, amt);
+  };
+
+  _PVector.add = function(v1, v2, target) {
+    return target? target.set(v1.x + v2.x, v1.y + v2.y, v1.z + v2.z)
+                 : new _PVector(v1.x + v2.x, v1.y + v2.y, v1.z + v2.z);
+  };
+
+  _PVector.sub = function(v1, v2, target) {
+    return target? target.set(v1.x - v2.x, v1.y - v2.y, v1.z - v2.z)
+                 : new _PVector(v1.x - v2.x, v1.y - v2.y, v1.z - v2.z);
+  };
+
+  _PVector.mult = function(v, n, target) {
+    if (typeof n === 'number') {
```

Comment:
Java and JS behave completely differently. Not wanting to instantiate an object is fine, but JS objects are completely different things compared to Java objects, so performance notions from Java are typically not portable to JS at all.